app/utils/callback/lava.py

```diff
@@ -0,0 +1,234 @@
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import bson
+import datetime
+import errno
+import json
+import models.boot
+import os
+import utils
+import yaml
+
+import utils.boot
+import utils.db
+import utils.lava_log_parser
+
+# copied from lava-server/lava_scheduler_app/models.py
+SUBMITTED = 0
+RUNNING = 1
+COMPLETE = 2
+INCOMPLETE = 3
+CANCELED = 4
+CANCELING = 5
+
+LAVA_JOB_RESULT = {
+    COMPLETE: 'PASS',
+    INCOMPLETE: 'FAIL',
```

Comment:
Nitpick: for consistency, please use double quotes, they have been used all around the code base.